src/sio.ts

```diff
@@ -33,11 +33,234 @@ const DIV_QUOTIENT = 0x070; //  Divider result quotient
+const INTERP1_POP_FULL = 0x0dc; // Read FULL result, and simultaneously write lane results to both accumulators (POP)
+const INTERP1_PEEK_LANE0 = 0x0e0; // Read LANE0 result, without altering any internal state (PEEK)
+const INTERP1_PEEK_LANE1 = 0x0e4; // Read LANE1 result, without altering any internal state (PEEK)
+const INTERP1_PEEK_FULL = 0x0e8; // Read FULL result, without altering any internal state (PEEK)
+const INTERP1_CTRL_LANE0 = 0x0ec; // Control register for lane 0
+const INTERP1_CTRL_LANE1 = 0x0f0; // Control register for lane 1
+const INTERP1_ACCUM0_ADD = 0x0f4; // Values written here are atomically added to ACCUM0
+const INTERP1_ACCUM1_ADD = 0x0f8; // Values written here are atomically added to ACCUM1
+const INTERP1_BASE_1AND0 = 0x0fc; // On write, the lower 16 bits go to BASE0, upper bits to BASE1 simultaneously
+
 //SPINLOCK
 const SPINLOCK_ST = 0x5c;
 const SPINLOCK0 = 0x100;
 const SPINLOCK31 = 0x17c;
 
+class InterpolatorConfig {
+  shift = 0;
+  mask_lsb = 0;
+  mask_msb = 0;
+  is_signed = false;
+  cross_input = false;
+  cross_result = false;
+  add_raw = false;
+  force_msb = 0;
+  blend = false;
+  clamp = false;
+  overf0 = false;
+  overf1 = false;
+  overf = false;
+  _reserved0 = 0;
+
+  constructor(value: number) {
+    this.shift        =         (value >>>  0) & 0b11111;
+    this.mask_lsb     =         (value >>>  5) & 0b11111;
+    this.mask_msb     =         (value >>> 10) & 0b11111;
+    this.is_signed    = Boolean((value >>> 15) & 1);
+    this.cross_input  = Boolean((value >>> 16) & 1);
+    this.cross_result = Boolean((value >>> 17) & 1);
+    this.add_raw      = Boolean((value >>> 18) & 1);
+    this.force_msb    =         (value >>> 19) & 0b11;
+    this.blend        = Boolean((value >>> 21) & 1);
+    this.clamp        = Boolean((value >>> 22) & 1);
+    this.overf0       = Boolean((value >>> 23) & 1);
+    this.overf1       = Boolean((value >>> 24) & 1);
+    this.overf        = Boolean((value >>> 25) & 1);
+    this._reserved0   =         (value >>> 26) & 0b111111;
+  }
+
+  toUint32() {
+    return (
+      ((this.shift                & 0b11111)  <<  0) |
+      ((this.mask_lsb             & 0b11111)  <<  5) |
+      ((this.mask_msb             & 0b11111)  << 10) |
+      ((Number(this.is_signed)    & 1)        << 15) |
+      ((Number(this.cross_input)  & 1)        << 16) |
+      ((Number(this.cross_result) & 1)        << 17) |
+      ((Number(this.add_raw)      & 1)        << 18) |
+      ((this.force_msb            & 0b11)     << 19) |
+      ((Number(this.blend)        & 1)        << 21) |
+      ((Number(this.clamp)        & 1)        << 22) |
+      ((Number(this.overf0)       & 1)        << 23) |
+      ((Number(this.overf1)       & 1)        << 24) |
+      ((Number(this.overf)        & 1)        << 25) |
+      ((this._reserved0           & 0b111111) << 26)
+    );
+  }
+}
+
+class BitUtil {
```

Comment:
Please move these two functions into [src/utils/bit.ts](https://github.com/wokwi/rp2040js/blob/master/src/utils/bit.ts), then import them from that file. Also, since they are totally pure and stateless, no need to wrap them in a class.

Reply:
will do!

Reply:
fixed